grapher/stackedCharts/StackedDiscreteBarChart.tsx

```diff
@@ -362,6 +379,209 @@ export class StackedDiscreteBarChart
+        const dp = Math.ceil(-Math.log10(point.value) + 1)
+        const barLabel = formatColumn.formatValueShort(point.value, {
+            numDecimalPlaces: isFinite(dp) && dp >= 0 ? dp : 0,
+        })
+        const labelBounds = Bounds.forText(barLabel, {
+            fontSize: 12.6,
+        })
+        // Check that we have enough space to show the bar label
+        const showLabelInsideBar =
+            labelBounds.width < 0.85 * barWidth &&
+            labelBounds.height < 0.85 * barHeight
+        const labelColor = isDarkColor(color) ? "#fff" : "#000"
+
+        return (
+            <TippyIfInteractive
+                lazy
+                isInteractive={!this.manager.isExportingtoSvgOrPng}
+                key={seriesName}
+                hideOnClick={false}
+                content={<StackedDiscreteBarChart.Tooltip {...tooltipProps} />}
+            >
+                <g>
+                    <rect
+                        x={0}
+                        y={0}
+                        transform={`translate(${barX}, ${-barHeight / 2})`}
+                        width={barWidth}
+                        height={barHeight}
+                        fill={color}
+                        opacity={isFaint ? 0.1 : 0.85}
+                        style={{
+                            transition: "height 200ms ease",
+                        }}
+                    />
+                    {showLabelInsideBar && (
+                        <text
+                            x={barX + barWidth / 2}
+                            y={0}
+                            width={barWidth}
+                            height={barHeight}
+                            fill={labelColor}
+                            opacity={isFaint ? 0 : 1}
+                            fontSize="0.7em"
+                            textAnchor="middle"
+                            dominantBaseline="middle"
+                        >
+                            {barLabel}
+                        </text>
+                    )}
+                </g>
+            </TippyIfInteractive>
+        )
+    }
+
+    private static Tooltip(props: TooltipProps) {
+        let hasTimeNotice = false
+
+        return (
+            <table
+                style={{
+                    lineHeight: "1em",
+                    whiteSpace: "normal",
+                    borderSpacing: "0.5em",
+                }}
+            >
+                <tbody>
+                    <tr>
+                        <td colSpan={4} style={{ color: "#111" }}>
+                            <strong>{props.label}</strong>
+                        </td>
+                    </tr>
+                    {props.bars.map((bar) => {
+                        const { highlightedSeriesName } = props
+                        const squareColor = bar.color
+                        const isHighlighted =
+                            bar.seriesName === highlightedSeriesName
+                        const isFaded =
```

Comment:
nit: `isFaded` and `isFaint` have the same intended meaning?

Reply:
Yeah, and they even sound so similar!